components/server/src/workspace/config-provider.ts

```diff
@@ -169,36 +171,33 @@ export class ConfigProvider {
                     customConfigString = content;
                     // We do not only care about the config itself but also where we got it from
                     configBasePath = basePath;
-                    fromDefinitelyGp = true;
+                    origin = "definitely-gp";
+                }
+
+                if (!customConfigString) {
+                    // if there's still no configuration, let's infer one
+                    customConfigString = await inferredConfig;
+                    origin = "derived";
                 }
 
                 if (customConfigString) {
                     const parseResult = this.gitpodParser.parse(customConfigString);
                     customConfig = parseResult.config
-                    await this.fillInDefaultLocations(customConfig, inferredConfig);
                     if (parseResult.validationErrors) {
                         const err = new InvalidGitpodYMLError(parseResult.validationErrors);
                         // this is not a system error but a user misconfiguration
                         log.info(logContext, err.message, { repoCloneUrl: commit.repository.cloneUrl, revision: commit.revision, customConfigString });
                         throw err;
                     }
-                    customConfig._origin = fromProjectDB ? 'project-db' : (fromDefinitelyGp ? 'definitely-gp' : 'repo');
-                } else {
-                    /* There is no configuration for this repository. Before we fall back to the default config,
-                    * let's see if there is language specific configuration we might want to apply.
-                    */
-                    customConfig = await inferredConfig;
-                    if (customConfig) {
-                        customConfig._origin = 'derived';
-                    }
+                    customConfig._origin = origin;
                 }
             }
 
             if (!customConfig) {
                 return undefined;
             }
 
-            return { customConfig, configBasePath };
+            return { customConfig, configBasePath, literalConfig: { '.gitpod.yml': customConfigString || ''} };
```

Comment:
**issue**: Is it expected that the new untracked `.gitpod.yml` is not visible as changelist in the workspaces list when the workspace stops?

| Workspaces with new auto-generated `.gitpod.yml` and one file change |
|-|
| <img width="1336" alt="Screenshot 2021-12-29 at 11 26 37 PM" src="https://user-images.githubusercontent.com/120486/147704644-73468953-4bbc-4d4b-8fcd-7195771a8826.png"> |

Reply:
This is pretty unrelated and should work as reliable as it does with other pending change.
Related issue https://github.com/gitpod-io/gitpod/issues/7370